src/plone/restapi/tests/test_resolveuid_functional.py

```diff
@@ -0,0 +1,114 @@
+
+    layer = PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer["portal"]
+        self.request = self.layer["request"]
+        login(self.portal, TEST_USER_NAME)
+        setRoles(self.portal, TEST_USER_ID, ["Manager"])
+        fti = DexterityFTI("blockspage")
+        self.portal.portal_types._setObject("blockspage", fti)
+        fti.klass = "plone.dexterity.content.Container"
+        fti.behaviors = ("volto.blocks",)
+        self.fti = fti
+        alsoProvides(self.request, IBlocks)
+        self.portal_url = self.portal.absolute_url()
+
+        self.api_session = RelativeSession(self.portal_url)
+        self.api_session.headers.update({"Accept": "application/json"})
+        self.api_session.auth = (SITE_OWNER_NAME, SITE_OWNER_PASSWORD)
+
+        self.portal.invokeFactory("Document", id="target", title="Target")
+        self.target = self.portal.target
+
+        transaction.commit()
+
+    def tearDown(self):
+        self.api_session.close()
+
+    def test_resolveuid(self):
+        response = self.api_session.post(
+            "/",
+            json={
+                "title": "Document",
+                "@type": "blockspage",
+                "blocks": {
+                    "ca5908a5-3f58-4cd5-beb7-9bd1539d6744": {"@type": "title"},
+                    "791bf004-7c88-4278-8490-13b85c3fa4b4": {
+                        "@type": "text",
+                        "text": {
+                            "blocks": [
+                                {
+                                    "key": "3bnq6",
+                                    "text": "Link",
+                                    "type": "unstyled",
+                                    "depth": 0,
+                                    "inlineStyleRanges": [],
+                                    "entityRanges": [
+                                        {"offset": 0, "length": 4, "key": 0}
+                                    ],
+                                    "data": {},
+                                }
+                            ],
+                            "entityMap": {
+                                "0": {
+                                    "type": "LINK",
+                                    "mutability": "MUTABLE",
+                                    "data": {
+                                        "url": "http://localhost:55001/Plone/target"
+                                    },
+                                }
+                            },
+                        },
+                    },
+                },
+                "blocks_layout": {
+                    "items": [
+                        "ca5908a5-3f58-4cd5-beb7-9bd1539d6744",
+                        "791bf004-7c88-4278-8490-13b85c3fa4b4",
+                    ]
+                },
+            },
+        )
+        transaction.commit()
+
+        self.assertEqual(
+            "http://localhost:55001/Plone/target",
```

Comment:
@sneridagh @buchi I'd expect this to be a resolveUID link or am I missing something obvious here? The serializer should convert the path to a resolveUID that is stored internally, correct? It is super odd that the serializer/deserializer tests work but nothing is actually stored in the database.  cc @wkbkhard

Reply:
@tisto FOUND IT!
Deserializer stores the value correctly, but searializer is actual overriding that value when it convert the uid into a path.

The reason could be that the we are working with a shallow copy of the field's value, and every change into it, will be reflected into the original one.
With a deep copy, everything works fine.

Reply:
@cekk w00t!!!

Reply:
@cekk would you mind pushing your solution to this branch?